File: PythonAlgorithmInterView/그래프/조합.py
def combine(self, n: int, k: int) -> list[list[int]]:

    result = []

    def dfs(elements, start, k):

        if k == 0 :
            result.append(elements[:])
            return

        for i in range( start , n+1):
            elements.append(i)
            dfs(elements, i +1, k -1)
            elements.pop()

    dfs([], 1, k)

    return result
if __name__ == '__main__':
    n = 4
    k = 2

    combine(None, n,k)

File: PythonAlgorithmInterView/그래프/조합.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number[][]}
 * 문제 77번
 */
 var combine = function(n, k) {
    
    let result = [];

    function dfs(elements, start, k){
        if(k == 0){
            result.push(elements.slice());
            return;
        }

        for(let i = start; i < n+1; i++){
            elements.push(i);
            dfs(elements, i+1, k-1);
            elements.pop();
        }
    }

    dfs([], 1, k);

    return result
};


let n = 4, k = 2;
console.log( combine(n,k))